### PR TITLE
Added error case at login if user email is not verified

### PR DESCRIPTION
**Pull request by charliescheer**
### Fix
This PR adds a new error case while logging in where a user's email address has not been verified they will receive an error alerting them to the fact that they can not login till their email is verified and then supplies them with a button to resend the verification email.

### Test
1. setup a tool like Charles proxy to get in the way of the login request response.
2. When the response returns, change the response value to 403
3. You should see an alert with two options, verify and okay.

Note this will require more testing when the backend changes are made to return the 403 response to an unverified email, currently that has not been added in, so it can not be fully tested.

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer is required to review these changes, but anyone can perform the review.

### CODE REVIEW

File: Simplenote/Classes/SPAuthError.swift [hunk 1]
```diff
@@ -23,6 +24,8 @@ extension SPAuthError {
         switch loginErrorCode {
         case 401:
             self = response == Constants.compromisedPassword ? .compromisedPassword : .loginBadCredentials
+        case 403:
+            self = .unverifiedEmail
```
Comment (eshurakov) [inline Simplenote/Classes/SPAuthError.swift]: I believe we also wanted to send an error code in the body (`requires verification`) similarly to how we did it with compromised password.
Let's check with @nsakaimbo

File: Simplenote/Classes/SPAuthViewController.swift [hunk 2]
```diff
@@ -443,6 +445,16 @@ private extension SPAuthViewController {
         present(alertController, animated: true, completion: nil)
     }
 
+    func presentUserUnverifiedError(error: SPAuthError) {
+        let alertController = UIAlertController(title: error.title, message: error.message, preferredStyle: .alert)
+        alertController.addCancelActionWithTitle(AuthenticationStrings.unverifiedCancelText)
+        alertController.addDefaultActionWithTitle(AuthenticationStrings.unverifiedActionText) { _ in
+            AccountRemote().verify(email: self.email) { _ in }
```
Comment (eshurakov) [inline Simplenote/Classes/SPAuthViewController.swift]: I think we should show an activity indicator (can use the one we have for account deletion) and handle errors similar to what we're doing in `AccountVerificationViewController`

File: RELEASE-NOTES.txt [hunk 3]
```diff
@@ -1,6 +1,7 @@
 4.43
 -----
 -   [Internal] added login alert for compromised password #1394
+- [Intenrnal] added error response case on login for unverified emails #1398
```
Comment (jleandroperez) [inline RELEASE-NOTES.txt]:
```suggestion
-   [Internal] added error response case on login for unverified emails #1398
```

File: Simplenote/Classes/SPAuthError.swift (hunk 1 above)
Comment (charliescheer) [reply Simplenote/Classes/SPAuthError.swift]: I chatted with @nsakaimbo during our one on one today and he agreed that we should also check for the requires verification body.  I have updated the changes to do that the same way we did in compromised password.

**Comment by charliescheer** [comment]
Thanks for those suggestions @eshurakov I have updated the codes.  This is ready for another look

File: Simplenote/Classes/UIAlertController+Helpers.swift [hunk 4]
```diff
@@ -26,4 +26,12 @@ extension UIAlertController {
 
         return action
     }
+
+    static func dismissableAlert(title: String,
+                                        message: String,
+                                        style: UIAlertController.Style = .alert) -> UIAlertController {
+        let alert = UIAlertController(title: title, message: message, preferredStyle: style)
+        alert.addCancelActionWithTitle("Okay")
```
Comment (eshurakov) [inline Simplenote/Classes/UIAlertController+Helpers.swift]: Not localized string :-)

File: Simplenote/Classes/SPAuthViewController.swift [hunk 5]
```diff
@@ -670,6 +697,12 @@ private enum AuthenticationStrings {
     static let loginActionText              = NSLocalizedString("Log In", comment: "Log In Action")
     static let compromisedAlertCancel       = NSLocalizedString("Not Yet", comment: "Cancel action for password alert")
     static let compromisedAlertReset        = NSLocalizedString("Change Password", comment: "Change password action")
+    static let unverifiedCancelText         = NSLocalizedString("Okay", comment: "Email unverified alert dismiss")
```
Comment (eshurakov) [inline Simplenote/Classes/SPAuthViewController.swift]: We never used "Okay" before, in all other alerts it says "OK"

**Review by eshurakov — APPROVED** [review]
Besides the fact that I find it strange to use alerts instead of reusing the UI that we already have for account verification and a few localization nitpicks, everything works as expected 👍

File: Simplenote/Classes/SPAuthError.swift [hunk 6]
```diff
@@ -23,6 +24,10 @@ extension SPAuthError {
         switch loginErrorCode {
         case 401:
             self = response == Constants.compromisedPassword ? .compromisedPassword : .loginBadCredentials
+        case 403:
+            self = response == Constants.requiresVerification ?
+                .unverifiedEmail :
+                .unknown(statusCode: loginErrorCode, response: response, error: error)
```
Comment (jleandroperez) [inline Simplenote/Classes/SPAuthError.swift]: BTW you can also use `where`:

```
    init(loginErrorCode: Int, response: String?, error: Error?) {
        switch loginErrorCode {
        case 401:
            self = response == Constants.compromisedPassword ? .compromisedPassword : .loginBadCredentials
        case 403 where response == Constants.requiresVerification:
            self = .unverifiedEmail
        default:
            self = .unknown(statusCode: loginErrorCode, response: response, error: error)
        }
    }
```

File: Simplenote/Classes/SPAuthViewController.swift [hunk 7]
```diff
@@ -443,6 +445,31 @@ private extension SPAuthViewController {
...
+
+            AccountRemote().verify(email: email) { result in
+                spinnerVC.dismiss(animated: false, completion: nil)
+                var alert: UIAlertController
+                switch result {
+                case .success:
+                    alert = UIAlertController.dismissableAlert(title: AuthenticationStrings.verificationSentTitle,
+                                                                   message: String(format: AuthenticationStrings.verificationSentTemplate, email))
+                case .failure:
+                    alert = UIAlertController.dismissableAlert(title: AuthenticationStrings.unverifriedErrorTitle,
```
Comment (jleandroperez) [inline Simplenote/Classes/SPAuthViewController.swift]: Nipticky: `unverifried` (fried!)

File: Simplenote/Classes/SPAuthViewController.swift [hunk 8]
```diff
@@ -443,6 +445,31 @@ private extension SPAuthViewController {
...
+            let spinnerVC = SpinnerViewController()
+            self?.present(spinnerVC, animated: false, completion: nil)
+
+            AccountRemote().verify(email: email) { result in
+                spinnerVC.dismiss(animated: false, completion: nil)
+                var alert: UIAlertController
+                switch result {
+                case .success:
+                    alert = UIAlertController.dismissableAlert(title: AuthenticationStrings.verificationSentTitle,
+                                                                   message: String(format: AuthenticationStrings.verificationSentTemplate, email))
```
Comment (jleandroperez) [inline Simplenote/Classes/SPAuthViewController.swift]: Nipticky: alignment!

File: Simplenote/Classes/SPAuthViewController.swift [hunk 9]
```diff
@@ -443,6 +445,31 @@ private extension SPAuthViewController {
...
+                case .failure:
+                    alert = UIAlertController.dismissableAlert(title: AuthenticationStrings.unverifriedErrorTitle,
+                                                                   message: AuthenticationStrings.unverifiedErrorMessage)
```
Comment (jleandroperez) [inline Simplenote/Classes/SPAuthViewController.swift]: Nipticky: Alignment!

**Review by jleandroperez — APPROVED** [review]
Couple nipticky notes. :shipit: when ready!!

File: Simplenote/Classes/SPAuthViewController.swift (hunk 7 above)
Comment (charliescheer) [reply Simplenote/Classes/SPAuthViewController.swift]: I was having some crazy typo issues yesterday!

**Pull request merged**